src/bootstrap/builder.rs

```diff
@@ -1718,6 +1718,19 @@ impl<'a> Builder<'a> {
             cargo.env("RUSTC_HOST_CRT_STATIC", x.to_string());
         }
 
+        // Users won't have the original cargo registry from the CI builder available, even if they have `rust-src` installed.
+        // Don't show their sources in UI tests even if they're available.
+        // As a happy side-effect, this fixes a few UI tests when download-rustc is enabled.
+        // NOTE: only set this when building std so the error messages are better for rustc itself.
+        if mode == Mode::Std {
+            #[allow(deprecated)]
+            // we do actually want /home/you on Mingw, since that's where cargo puts the registry
```

Comment:
I don't think this is the case. With MinGW (`*-pc-windows-gnu` triple) Cargo will store its files under `$USERPROFILE`:
```
### MSYS2 shell
$ file $HOME/.cargo
/home/mateusz/.cargo: cannot open `/home/mateusz/.cargo' (No such file or directory)

$ file $USERPROFILE/.cargo
C:\Users\mateusz/.cargo: directory

### Git for Windows shell
$ file $HOME/.cargo
/c/Users/mateusz/.cargo: directory

$ file $USERPROFILE/.cargo
C:\Users\mateusz/.cargo: directory
```

Reply:
you were absolutely right and using USERPROFILE fixed CI :) thank you!

Reply:
Glad it helped ;)